src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,137 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final Object[] EMPTY_LIST = {};
+    private Object[] elementArray;
+    private int size;
+
+    public ArrayList() {
+        this.elementArray = EMPTY_LIST;
+    }
+
+    private Object[] grow(int minCapacity) {
+        int oldCapacity = elementArray.length;
+        int newCapacity = minCapacity + (oldCapacity >> 1);
+        if (oldCapacity > 0) {
+            return elementArray = Arrays.copyOf(elementArray, newCapacity);
+        } else {
+            return elementArray = new Object[Math.max(DEFAULT_CAPACITY, newCapacity)];
+        }
+    }
+
+    private Object[] grow() {
+        return grow(size + 1);
+    }
 
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
-
+        rangeCheck(index);
+        if (index == elementArray.length) {
+            elementArray = grow();
+        }
+        if (index < size) {
+            Object[] tempArray = new Object[size - index];
+            System.arraycopy(elementArray, index, tempArray, 0, size - index);
+            elementArray[index] = value;
+            System.arraycopy(tempArray, 0, elementArray, index + 1, tempArray.length);
+        } else {
+            elementArray[index] = value;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            if (size + list.size() >= elementArray.length) {
+                grow();
+            }
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index > size - 1 || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index is out of bounds");
+        }
+        return (T) elementArray[index];
     }
 
     @Override
     public void set(T value, int index) {
```

Comment:
use rangeCheck here

Reply:
this comment is not fixed 
![image](https://user-images.githubusercontent.com/58385506/114024326-95b58500-987c-11eb-9307-e4b77c22e12b.png)